asv_bench/benchmarks/multiindex_object.py

```diff
@@ -160,4 +160,43 @@ def time_equals_non_object_index(self):
         self.mi_large_slow.equals(self.idx_non_object)
 
 
+class SetOperations:
+
+    params = [
+        ("monotonic", "non_monotonic"),
+        ("datetime", "int", "string"),
+        ("intersection", "union", "symmetric_difference"),
+    ]
+    param_names = ["index_structure", "dtype", "method"]
+
+    def setup(self, index_structure, dtype, method):
+        N = 10 ** 5
```

Comment:
it looks like N is only used for N // 1000, why not just define N = 100?

Reply:
I wanted to make it obvious that `MultiIndex.from_product([level1, level2])` should result in an index of size N since 1000 is the size of `level1`.